### DISCUSSION TITLE
[firebase_crashlytics] add `instanceFor(FirebaseApp app)` method in FirebaseCrashlytics

### DISCUSSION THREAD

**Discussion by wferem1** [firebase/flutterfire#8920] [Feature request]
I'm building an application where we have different backend stages (development, testing, production, ...), which I want to access and switch to in the application. 
For each of these stages, there is a firebase project, and when we switch stage, I want to switch to that firebase project for crashlytics, so that we can keep the error reporting separated for these stages.

My proposal is to add the static method to FirebaseCrashlytics
```dart
static FirebaseCrashlytics instanceFor({
    FirebaseApp? app,
 })
```
so that I can specify which Firebase project I can send Error logs to

**Comment by TamasBarta** [comment]
There is a response to this in #9109, however they state that the native counterparts don't support this.

They truly don't, but I checked the source code, and I have a question about it. Here's the `getInstance` I found on the native Android `FirebaseCrashlytics`:

```java
@NonNull
public static FirebaseCrashlytics getInstance() {
    FirebaseApp app = FirebaseApp.getInstance();
    FirebaseCrashlytics instance = (FirebaseCrashlytics)app.get(FirebaseCrashlytics.class);
    if (instance == null) {
        throw new NullPointerException("FirebaseCrashlytics component is not present.");
    } else {
        return instance;
    }
}
```

I see that this class only has this `getInstance` method to get a `FirebaseCrashlytics`, but it isn't hard to see that the app itself could very well be a parameter. @russellwheatley, I saw you closed the issue, so I assume you are well-equipped to answer this, but do you think this method to obtain a `FirebaseCrashlytics` could be used from outside `FirebaseCrashlytics` (in the Flutter plugins) until the native libraries are fixed as well? (I'm not aware of an upstream bug report)

I can't tell if it would be a good idea considering the native libraries' behaviour, but I believe at least Flutter devs could still throw the JSON and plist horrors out, and decide Flutter apps at runtime.
